test/drbgtest.c

```diff
@@ -615,7 +616,8 @@ static int test_drbg_reseed(int expect_success,
      */
 
     /* Generate random output from the public and private DRBG */
-    before_reseed = expect_master_reseed == 1 ? time(NULL) : 0;
+    before_reseed = reseed_time ? reseed_time
+                                : expect_master_reseed == 1 ? time(NULL) : 0;
```

Comment:
This nested conditional expression is incorrect, because `before_reseed` time should only be set to a nonzero value if  `expect_master_reseed` equals 1. The correct expression would be

```C
    before_reseed = expect_master_reseed == 1 ? 
                         (reseed_time ? reseed_time : time(NULL)) : 0;
```

although the incorrectness never shows up in practice because the only time when a nonzero `reseed_time` value is passed, the `expect_master_reseed` value also equals 1.

However, there is a much simpler solution which requires less mind acrobatics:


```C
    /*
     * step 2: generate random output
     */
     
+   if (reseed_time == 0)
+       reseed_time = time(NULL);

    /* Generate random output from the public and private DRBG */
    before_reseed = expect_master_reseed == 1 ? reseed_time : 0;
    if (!TEST_int_eq(RAND_bytes(buf, sizeof(buf)), expect_success)
        || !TEST_int_eq(RAND_priv_bytes(buf, sizeof(buf)), expect_success))
        return 0;
    after_reseed = time(NULL);
```

Reply:
yeah, I pondered back and forth over the possible ways to do this initialization,
and I did not like using a input parameter as a variable either....